lib/msal-browser/src/interaction_client/WamInteractionClient.ts

```diff
@@ -0,0 +1,174 @@
+
+        const uid = response.account.properties["UID"] || idTokenObj.claims.oid || idTokenObj.claims.sub || Constants.EMPTY_STRING;
+        const tid = response.account.properties["TenantId"] || idTokenObj.claims.tid || Constants.EMPTY_STRING;
+
+        const result: AuthenticationResult = {
+            authority: request.authority,
+            uniqueId: uid,
+            tenantId: tid,
+            scopes: responseScopes.asArray(),
+            account: accountEntity.getAccountInfo(),
+            idToken: response.id_token,
+            idTokenClaims: idTokenObj.claims,
+            accessToken: response.access_token,
+            fromCache: false,
+            expiresOn: new Date(Number(reqTimestamp + response.expires_in) * 1000),
+            tokenType: AuthenticationScheme.BEARER,
+            correlationId: this.correlationId,
+            state: response.state
+        };
+
+        return result;
+    }
+
+    /**
+     * Validates WAM response before processing
+     * @param response 
+     */
+    private validateWamResponse(response: object): void {
+        if (
+            response.hasOwnProperty("access_token") &&
+            response.hasOwnProperty("id_token") &&
+            response.hasOwnProperty("client_info") &&
+            response.hasOwnProperty("account") &&
+            response.hasOwnProperty("scopes") &&
+            response.hasOwnProperty("expires_in")
+        ) {
+            return;
+        } else {
+            throw WamAuthError.createUnexpectedError("Response missing expected properties.");
+        }
+    }
+
+    /**
+     * Translates developer provided request object into WamRequest object
+     * @param request 
+     */
+    protected initializeWamRequest(request: PopupRequest|SsoSilentRequest): WamTokenRequest {
+        this.logger.trace("WamInteractionClient - initializeWamRequest called");
+
+        if (request.authenticationScheme && request.authenticationScheme !== AuthenticationScheme.BEARER) {
+            // Only Bearer flows are supported right now
+            throw WamAuthError.createWamAtPopNotSupportedError();
+        }
+
+        const authority = request.authority || this.config.auth.authority;
+        const canonicalAuthority = new UrlString(authority);
+        canonicalAuthority.validateAsUri();
+        
+        const scopes = request && request.scopes || [];
+        const scopeSet = new ScopeSet(scopes);
+        scopeSet.appendScopes(OIDC_DEFAULT_SCOPES);
+
+        const validatedRequest: WamTokenRequest = {
+            ...request,
+            clientId: this.config.auth.clientId,
+            authority: canonicalAuthority.urlString,
+            scopes: scopeSet.printScopes(),
+            redirectUri: this.getRedirectUri(request.redirectUri),
+            correlationId: this.correlationId,
+            prompt: request.prompt || PromptValue.NONE,
+            instanceAware: false,
```

Comment:
Currently an issue with the WAM implementation blocking this feature. Will make this configurable in a follow up PR when that gets resolved